firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java

```diff
@@ -531,11 +528,7 @@ private ListenerRegistration addSnapshotListenerInternal(
         new AsyncEventListener<>(userExecutor, viewListener);
 
     com.google.firebase.firestore.core.Query query = asQuery();
-    QueryListener queryListener = firestore.getClient().listen(query, options, asyncListener);
-
-    return ActivityScope.bind(
-        activity,
-        new ListenerRegistrationImpl(firestore.getClient(), queryListener, asyncListener));
+    return firestore.callClient(client -> client.listen(query, options, activity, asyncListener));
```

Comment:
Can we keep the code around activity in this method instead of pushing it down?

Reply:
Fine with me. Fixing throughout.